.github/workflows/ci.yml

```diff
@@ -102,7 +103,8 @@ jobs:
     name: target (${{ matrix.pretty }},${{ matrix.os }})
     runs-on: ${{ matrix.os }}
     needs: [shellcheck, test, generate-matrix, check]
-    if: github.event_name == 'push'
+    if: false
+    #if: github.event_name == 'push' && ${{ needs.generate-matrix.outputs.matrix != '{}' && needs.generate-matrix.outputs.matrix != '[]' && needs.generate-matrix.outputs.matrix != '' }}
```

Comment:
I think my logic is invalid here, since just skipping the `build` step altogether ensures the empty matrix doesn't occur.

Reply:
the problem I think is that

`conclusions` is `if: always()`, and in its needs, `build` is included. if `build` is skipped, but has elements in it's matrix, then it will run, if its matrix is empty and not skipped, it will fail (for some reason)

best way to solve this is to make conclusion see if `needs.generate-matrix.outputs.matrix` is empty, and if it is mark `build` as succeeded (on try only)

Reply:
That's why my logic is off: the build step is supposed to be skipped if the matrix is empty.